### DISCUSSION TITLE
intercepts not catching cy.requests?

### DISCUSSION THREAD

**Discussion by stephenpmurray** [cypress-io/cypress#14631] [Component Testing] [answered]
I want to do the following:
1. Create an intercept for a particular http graphql request, which sends a stubbed reply.
2. Send this graphql request with cy.request, and recieve the stubbed reply, using cy.wait and an alias

However, it appears that the request is completed before the cy.wait() method is called, and so i can't catch the request.
Is there a workaround for this?

Here is my example code:
```
    cy.intercept("POST", "/query", (req) => {
      if (req.body.operationName.includes("Devices")) {
        req.alias = "tester";
        req.reply({ body: { somefield: "asdf" } });
      }
    });

    cy.request({
        method: "POST",
        url: "/query",
        body: { operationName: "Devices",
          query: `query Devices { devices { devices {id, status }}}`,
        },
      })

    cy.wait('@tester').its("response.body").should("include", { somefield: "asdf" });

```

**Comment by bahmutov** [comment, marked as answer]
Yes, `cy.request` is a Cypress command, it executes a request that bypassed `cy.route` and `cy.intercept` handlers (those handlers are only for dealing with the requests sent by the app). I will update the documentation.

**Reply by stephenpmurray** [reply]
Thanks for the response. Is there a reason for bypassing the app? 

Just curious :)

**Reply by bahmutov** [reply]
Yes - `cy.request` was made as an "escape" valve for your tests to communicate with an external server. Thus `cy.request` for example is made from Node so it has no cors restrictions, etc. Mixing it with `cy.intercept` would make everything complex.

**Comment by bahmutov** [comment]
Added note to clarify to `cy.intercept` page https://github.com/cypress-io/cypress-documentation/pull/3495/
